source/common/memory/stats.cc

```diff
@@ -2,121 +2,179 @@
 
 #include <cstdint>
 
+#include "source/common/common/assert.h"
 #include "source/common/common/logger.h"
 
 #if defined(TCMALLOC)
-
 #include "tcmalloc/malloc_extension.h"
+#elif defined(GPERFTOOLS_TCMALLOC)
+#include "gperftools/malloc_extension.h"
+#endif
 
 namespace Envoy {
 namespace Memory {
 
 uint64_t Stats::totalCurrentlyAllocated() {
+#if defined(TCMALLOC)
   return tcmalloc::MallocExtension::GetNumericProperty("generic.current_allocated_bytes")
       .value_or(0);
+#elif defined(GPERFTOOLS_TCMALLOC)
```

Comment:
out of curiosity (this define predates this PR) do we have CI coverage of both these paths?

Reply:
nope, our coverage builds only run with gperf tcmalloc these days. So code path for tcmalloc is not executed in coverage checks.

Reply:
i tried to switch the allocator type in coverage ci step, it failed due to "non-symmetric" code paths for gperf tcmalloc and tcmalloc. Ideally, we should run 2 coverage builds for each allocator and dont consider other allocator code path in the coverage results.

Reply:
sorry I didn't mean coverage coverage, I meant general testing on one of the CI builds.  I just want to make sure this code is actually compiled and tested somewhere.

Reply:
We have 2 CI steps that use gperf tcmalloc:  [coverage](https://github.com/envoyproxy/envoy/blob/main/.bazelrc#L245) and [compile-time-options](https://github.com/envoyproxy/envoy/blob/main/.bazelrc#L370)